### DISCUSSION TITLE
GPLv3+ → AGPLv3+ license

### DISCUSSION THREAD

**Discussion by comradekingu** [WeblateOrg/weblate#4705] [Ideas]
GPLv3+ doensn't have the "network protection" clause which renders the function of the license invalid as it it is applied upon "distribution". For reasons that are solved in AGPLv3, software intended to be used as-a-service is then no different from software distributed as packages.

> Notwithstanding any other provision of this License, if you modify the Program, your modified version must prominently offer all users interacting with it remotely through a computer network (if your version supports such interaction) an opportunity to receive the Corresponding Source of your version by providing access to the Corresponding Source from a network server at no charge, through some standard or customary means of facilitating copying of software.

**Comment by nijel** [comment]
I don't see strong reason to do so:

* It requires additional work to ensure compliance with the AGPL (something to download the source code)
* IANAL, but I think it doesn't cover additional modules loaded into Weblate, so users would be still able to extend Weblate without releasing source code
* Publishing the source code doesn't automatically mean it will get merged and improve existing code base

**Comment by comradekingu** [comment]
@nijel 

- 1. GPLv3 already has this requirement, which this project is already under, on a platform that already does that just fine. This software already points to its source code as an extension of that. For anyone wanting to fork it, that is the actually easy part.
- 2. That is incorrect, as GPLv3+ is already strong and full copyleft. IANAL seconded. If I grant you that you are correct, you are still arguing I have a point wrt. the "platform" code.
- 3. I am not saying making changes to the source code necessarily means improvement, what I am saying is that providing the changes should one do so, is a great safeguard in making sure any feasible improvement makes it way into whatever code-base happens to represent "improvement" freely.
Edit: It is also quite a motivator to get ~reproducible builds improvements for regular users.

TL;DR The closed loophole is just a quirk that has been fixed, GPL represents the same requirements for literally any other type of software. More work is less work. That is what it is predicated on anyways.

Actual TL;DR
Extra work for those only hosting their instance: 0
Extra work for those changing the source code in doing so: Same requirements as every other GPLv3 software that happens to not be distributed over a network in its function.

**Reply by nijel** [reply]
> GPLv3 already has this requirement, which this project is already under, on a platform that already does that just fine.

GPLv3 requires distributing source code if you distribute the program. AGPLv3 requires providing it in application. That is definitely something that would have to be implemented. And it would have to be implemented by us to be available on Hosted Weblate as well. That seems to be non-trivial amount of work and I can see more useful things to implement than this.

> That is incorrect, as GPLv3+ is already strong and full copyleft. IANAL seconded. If I grant you that you are correct, you are still arguing I have a point wrt. the "platform" code.

Still that doesn't prevent anybody implementing extension to Weblate under different license. Listening to Django signal can hardly be considered as a derivative work.
